functional/FunctionalTest.java

```diff
@@ -3823,12 +3836,70 @@ public static Process runMinio() throws Exception {
     pb.redirectErrorStream(true);
     pb.redirectOutput(ProcessBuilder.Redirect.to(new File(MINIO_BINARY + ".log")));
 
-    System.out.println("starting minio server");
+    if (tls) {
+      System.out.println("starting minio server in TLS");
+    } else {
+      System.out.println("starting minio server");
+    }
     Process p = pb.start();
     Thread.sleep(10 * 1000); // wait for 10 seconds to do real start.
     return p;
   }
 
+  public static void runEndpointTests(boolean automated) throws Exception {
+    client = MinioClient.builder().endpoint(endpoint).credentials(accessKey, secretKey).build();
+    MinioAdminClient adminClient =
+        MinioAdminClient.builder().endpoint(endpoint).credentials(accessKey, secretKey).build();
+    adminClientTests = new TestMinioAdminClient(adminClient, mintEnv);
+    // Enable trace for debugging.
+    // client.traceOn(System.out);
+    if (!mintEnv) System.out.println(">>> Running tests:");
+    FunctionalTest.runTests();
+
+    if (automated) {
+      // Run TLS tests on TLS endpoint
+      client =
+          MinioClient.builder().endpoint(endpointTLS).credentials(accessKey, secretKey).build();
+      client.ignoreCertCheck();
+      adminClient =
+          MinioAdminClient.builder()
+              .endpoint(endpointTLS)
+              .credentials(accessKey, secretKey)
+              .httpClient(getUnsafeOkHttpClient())
```

Comment:
You could remove this and use `adminClient.ignoreCertCheck()`

Reply:
That doesn't work @balamurugana

Reply:
It seems `MinioAdminClient` has implementation problem. I will fix the issue in different PR